src/fu-engine.c

```diff
@@ -797,8 +821,19 @@ fu_engine_modify_config(FuEngine *self,
 			return FALSE;
 		}
 
-		/* modify, effective next reboot */
-		return fu_config_set_value(FU_CONFIG(self->config), section, key, value, error);
+		/* many options need a reboot after this */
+		if (!fu_config_set_value(FU_CONFIG(self->config), section, key, value, error))
+			return FALSE;
+
+		/* reload remotes */
+		if (g_strcmp0(key, "TestDevices") == 0) {
+			g_object_unref(self->remote_list);
+			self->remote_list = fu_remote_list_new();
+			if (!fu_engine_load_remotes(self, error))
+				return FALSE;
```

Comment:
this chunk feels like it's going to cause problems; all FwupdRemote's in the first list are now ~invalid. Can we add the flag, invalidate the "loaded" boolean and call fu_engine_load_remotes() to get the new list perhaps?

Reply:
Fixed